File: glitch/backend/app/schema/summary.py
from pydantic import BaseModel
from typing import Optional


class SummaryItem(BaseModel):
    rid: int
    task_risk: int
    task_count_idle: int
    task_count_run: int
    task_count_alert: int
    task_count_review: int
    task_count_complete: int
    task_count_total: int
    task_workload_total: int
    task_number_completed: int
    task_number_total: int
    bug_risk: int
    bug_count_idle: int
    bug_count_run: int
    bug_count_alert: int
    bug_count_review: int
    bug_count_complete: int
    bug_count_total: int
    bug_workload_total: int
    date_entry: str

    class Config:
        orm_mode = True


class SummaryUser(BaseModel):
    rid: int
    id_project: int
    rid_users: int
    name: Optional[str] = None
    task_risk: int
    task_count_idle: int
    task_count_run: int
    task_count_alert: int
    task_count_review: int
    task_count_complete: int
    task_count_total: int
    task_workload_total: int
    task_number_completed: int
    task_number_total: int
    bug_risk: int
    bug_count_idle: int
    bug_count_run: int
    bug_count_alert: int
    bug_count_review: int
    bug_count_complete: int
    bug_count_total: int
    bug_workload_total: int
    date_entry: str

    class Config:
        orm_mode = True

File: glitch/frontend/app/glitch/src/types/Summary.ts
export interface SummaryItem {
  rid: number
  task_risk: number
  task_count_idle: number
  task_count_run: number
  task_count_alert: number
  task_count_review: number
  task_count_complete: number
  task_count_total: number
  task_workload_total: number
  task_number_completed: number
  task_number_total: number
  bug_risk: number
  bug_count_idle: number
  bug_count_run: number
  bug_count_alert: number
  bug_count_review: number
  bug_count_complete: number
  bug_count_total: number
  bug_workload_total: number
  date_entry: string
}

export interface SummaryUser {
  rid: number
  id_project: number
  rid_users: number
  name: string
  task_risk: number
  task_count_idle: number
  task_count_run: number
  task_count_alert: number
  task_count_review: number
  task_count_complete: number
  task_count_total: number
  task_workload_total: number
  task_number_completed: number
  task_number_total: number
  bug_risk: number
  bug_count_idle: number
  bug_count_run: number
  bug_count_alert: number
  bug_count_review: number
  bug_count_complete: number
  bug_count_total: number
  bug_workload_total: number
  date_entry: string
}
